openpype/hosts/maya/api/lib.py

```diff
@@ -2639,6 +2649,39 @@ def _get_attrs(node):
     return world_space_nodes
 
 
+def _remove_scale_keys(node):
+    """Reset keys on scale attributes on source node.
+
+    Constraint parenting doesn't work with scale attributes. This method
+    removes keys, they need to be set again after constraint parenting.
+    """
+    scale_attrs = set(["sx", "sy", "sz"])
+
+    saved_keys = []
+    orig_key_value = [1.0]
+    key_value_changed = False
+    for attr in scale_attrs:
+        orig_node_attr = '{0}.{1}'.format(node, attr)
+        frames = cmds.keyframe(orig_node_attr, query=True) or []
```

Comment:
You are likely better of doing this since you'd be doing less calls against maya's slow api:
```python
times = cmds.keyframe(orig_node_attr, query=True, timeChange=True)
if not times:
    continue
values = cmds.keyframe(orig_node_attr, query=True, valueChange=True)
time_values = {time: value for time, value in zip(times, values)}
```

Reply:
Updated/cleaned up logic.